core/src/main/java/aima/core/search/api/DefinedGoalStatesProblem.java

```diff
@@ -0,0 +1,11 @@
+package aima.core.search.api;
+
+import java.util.List;
+
+/**
+ * we need a defined goal states for bidirectional problems
+ * @author wormi
+ */
+public interface DefinedGoalStatesProblem<A, S> extends Problem<A, S> {
```

Comment:
saw it and forgot about it :) I just copy the BidirectionalProblem and update my code accordingly